hathach/tinyusb
# PIO USB support

**Describe the PR**
Add support for [pico-pio-usb](https://github.com/sekigon-gonnoc/Pico-PIO-USB), allow TinyUSB can be used with both native usb (rhport0) and pio-usb (rhport1). Currently pio-usb and native usb must be on different mode. For recommendation, native should be device and pio-usb as host, since device has an stricter timing constraint, while in host mode, rp2040 basically control all the timing.

To test this PR
- check out this branch, and submodule lib/Pico-PIO-USB
- compile example host/hid_to_cdc, which is configured to use native usb as cdc device, and pio usb as host to get hid input. 
- any input event e.g mouse move, button will be printed to cdc on the native usb.

requires https://github.com/sekigon-gonnoc/Pico-PIO-USB/pull/22
